src/main/java/com/sproutsocial/nsq/Publisher.java

```diff
@@ -125,7 +125,12 @@ public synchronized void publish(String topic, List<byte[]> dataList) {
         } catch (Exception e) {
             logger.error("publish error", e);
             connectionDetails.markFailure();
-            publish(topic, dataList);
+            // We publish sequentially when we have an MPUB failure to
+            // help cover cases where perhaps the total payload size of the
+            // MPUB exceeded the nsqd -max-body-size or -max-msg-size.
+            for (final byte[] data : dataList) {
+                publish(topic, data);
```

Comment:
Nit: What happens if this singular publish throws an exception?  Should we wrap it in a catch, save an exception, and throw it at the end.  That way we give all of them a chance to be sent.